joplin/base/views/joplin_search_views.py

```diff
@@ -23,19 +25,49 @@
  - excluding certain pages from the pages to query
 """
 
+
+def dept_explorable_pages(user):
+    '''
+    In the spirit of UserPagePermissionsProxy's explorable_pages, check a user's permissions, excluding editor or
+    moderator permissions since those cast a wide net
+    https://github.com/wagtail/wagtail/blob/982b1d60a4b3d5b4e841f2b39e2244bcb421b091/wagtail/core/models.py#L1853
+    :param user: user making request
+    :return: Queryset of explorable pages
+    '''
+    # Deal with the trivial cases first...
+    if not user.is_active:
+        return Page.objects.none()
+    if user.is_superuser:
+        return Page.objects.all()  # todo: check this, we don't want the home page returned
+
+    user_perms = UserPagePermissionsProxy(user)
+    explorable_pages = Page.objects.none()
+
+    # exclude the moderator and editor groups
+    for perm in user_perms.permissions.filter(~Q(group_id=1) & ~Q(group_id=2)):
+        explorable_pages |= Page.objects.descendant_of(
+            perm.page, inclusive=True
+        )
+
+    return explorable_pages
+
+
 @vary_on_headers('X-Requested-With')
 @user_passes_test(user_has_any_page_permission)
 def search(request):
     # excluding wagtail 'page' pages and 'HomePages' from search (like home/root)
     homepage_content_type_id = ContentType.objects.get(app_label="home_page", model="homepage").id
-    pages = all_pages = Page.objects.all().exclude(content_type_id__in=[1, homepage_content_type_id]).prefetch_related('content_type').specific()
+    pages = all_pages = (
+        Page.objects.all().exclude(content_type_id__in=[1, homepage_content_type_id]).prefetch_related('content_type').specific()
+        & dept_explorable_pages(request.user))
```

Comment:
What does the "&" do here? It looks like we're fetching all pages for a content type and then we're fetching pages that a user can view?

Reply:
It's how wagtail handles it here: https://github.com/wagtail/wagtail/blob/92e9e156839e884d16c80d72f27baa27684c659f/wagtail/admin/views/pages.py#L68

the way I understand it, is its a union of the two querysets

Reply:
the & is more Q goodness right? https://docs.djangoproject.com/en/3.0/ref/models/querysets/#and

Nice!